File: seleniumEasyJava/src/test/java/E2E/poms/checkbox/CheckboxPOM.java
package E2E.poms.checkbox;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class CheckboxPOM {

    private WebDriver webDriver;

    public CheckboxPOM(WebDriver webDriver) {
        this.webDriver = webDriver;

        PageFactory.initElements(webDriver, this);
    }

    // ------------------- Single Checkbox Field --------------------------------

    @FindBy(id = "isAgeSelected")
    public WebElement singleCheckbox;

    @FindBy(id = "txtAge")
    public WebElement hiddenSuccessMessage;

    // ------------------- Multiple Checkbox Fields --------------------------------

    @FindBy(id = "check1")
    public WebElement checkboxButton;
}

File: seleniumEasyPython/poms/checkbox_POM.py
from selenium.webdriver.chrome.webdriver import WebDriver
from selenium.webdriver.common.by import By
from selenium.webdriver.remote.webelement import WebElement


class CheckboxPOM:
    def __init__(self, driver: WebDriver):
        self.driver = driver

    # ------------------- Single Checkbox Field --------------------------------

    def single_checkbox(self):
        element: WebElement = self.driver.find_element(By.ID, "isAgeSelected")
        return element

    def hidden_success_message(self):
        element: WebElement = self.driver.find_element(By.ID, "txtAge")
        return element

    # ------------------- Multiple Checkbox Fields --------------------------------

    def checkbox_1(self):
        element: WebElement = self.driver.find_element(By.XPATH,
                                                       '//*[@id="easycont"]/div/div[2]/div[2]/div[2]/div[1]/label'
                                                       '/input')
        return element

    def checkbox_2(self):
        element: WebElement = self.driver.find_element(By.XPATH,
                                                       '//*[@id="easycont"]/div/div[2]/div[2]/div[2]/div[2]/label'
                                                       '/input')
        return element

    def checkbox_3(self):
        element: WebElement = self.driver.find_element(By.XPATH, '//*[@id="easycont"]/div/div[2]/div[2]/div[2]/div[3]'
                                                                 '/label/input')
        return element

    def checkbox_4(self):
        element: WebElement = self.driver.find_element(By.XPATH, '//*[@id="easycont"]/div/div[2]/div[2]/div[2]/div[4]'
                                                                 '/label/input')
        return element

    def checkbox_button(self):
        element: WebElement = self.driver.find_element(By.ID, "check1")
        return element
